scripts/npm-audit-comparison.sh

```diff
@@ -0,0 +1,83 @@
+#!/bin/bash
+set -o errexit
+set -o pipefail
+set -o nounset
+set -o xtrace
+
+# Run npm audit on current branch and compare it with the results of running npm audit on the base branch that is set in the package.json. If there are more
+# high or critical findings in the current branch, then the test fails. If the same number of findings are found, then check that the vulnerabilities are the same.
+# If they are are different, then the test fails.
+# Test is optional to merge. Use your and the reviewer's best judgement
+
+# Save high and critical vulns from current branch
+npm ci
+npm audit | grep -E "(High)" -B3 -A10 > current-branch-high-vulnerabilities.txt || true
+npm audit | grep -E "(Critical)" -B3 -A10 > current-branch-critical-vulnerabilities.txt || true
+
+HIGH_VULN="$(grep -o High current-branch-high-vulnerabilities.txt | wc -l || true)"
+CRITICAL_VULN="$(grep -o Critical current-branch-critical-vulnerabilities.txt | wc -l || true)"
+echo $HIGH_VULN > compare-num-vulnerabilities.txt
+echo $CRITICAL_VULN >> compare-num-vulnerabilities.txt
+
+git stash
+
+# Save high and critical vulns from the base branch (e.g. develop, hotfix/1.11.2)
+git checkout "$npm_package_config_base_branch"
+npm ci
+npm audit | grep -E "(High)" -B3 -A10 > base-branch-high-vulnerabilities.txt || true
+npm audit | grep -E "(Critical)" -B3 -A10 > base-branch-critical-vulnerabilities.txt || true
+
+HIGH_VULN="$(grep -o High base-branch-high-vulnerabilities.txt | wc -l || true)"
```

Comment:
unsure about something, might have missed something might be worth a code comment. 
if all that `HIGH_VULN` and `CRITICAL_VULN` are storing are line counts
... ditto for the 'current_branch` / `base_branch` counts, how does this deal with the ticket example of removing one vulnerability but adding a different one?

Reply:
Yeah, so lines 14-15 save truncated version of the npm audit report to files. The truncated version has info about each vuln. Then just the count from the truncated version gets saved separately in new files (line 32-33). Further down I compare the counts first. If the current branch has a larger number of vulns, then it fails. But after that comparison, there's another check that compares the content of the files created from lines 14-15 if the count between the base and current branches are equal. Does that make sense?

Reply:
Ah, I missed the `cmp`, so its a "compare the number, but if equal diff/cmp to see if there is movement"
works for me, thanks!